ComfortZoneApp/app/src/main/java/com/example/comfortzone/WeatherClient.java

```diff
@@ -20,4 +30,28 @@ public String getWeatherURL(String lat, String lon) {
 
         return url;
     }
+
+    public String getWeatherData(String lat, String lon) {
+        String url = getWeatherURL(lat, lon);
+        final String[] data = new String[1];
+        final Request request = new Request.Builder()
+                .url(url)
+                .build();
+        newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(@NonNull Call call, @NonNull IOException e) {
+                Log.e(TAG, "failed making api request " + e);
+            }
+
+            @Override
+            public void onResponse(@NonNull Call call, @NonNull Response response) throws IOException {
+                try {
+                    data[0] = response.body().string();
+                } catch (IOException e) {
+                    Log.e(TAG, "could not get body" + e);
+                }
+            }
+        });
+        return data[0];
```

Comment:
How will this work? since you will be making network call, you won't get the data until it returns. Maybe you will need to pass in a callback